### ISSUE TITLE
Slow connectivity causes duplicate messages

### ISSUE THREAD

**Issue by arcturial** [lukefx/hubot-telegram#55]
I think we should change the polling mechanism from `setInterval` to `setTimeout`

If the bot experiences slow internet connectivity, the `setInterval` function will trigger the callback multiple times before the first callback is complete and the message offset is updated. This results in the same messages being handled multiple times.

If we change this to be `setTimeout` and recreate a new timeout once the old one completes, this problem should not occur anymore.

The bot will then effectively "lag" until the `getUpdates` call is complete or fails. Thoughts, @lukefx?

**Comment by kiliankoe** [comment]
I'm experiencing the same issue 😕

**Comment by waishing** [comment]
I fixed it by
```coffee
    @offset = 0
    @lastoffset = 0
.
.
.

setInterval ->
        self.api.invoke 'getUpdates', {offset: self.getLastOffset(), limit: 10}, (err, result) ->
          if (err)
            self.emit 'error', err
          else
            self.offset = result[result.length - 1].update_id if result.length

            for msg in result
              self.handleUpdate msg if self.lastoffset != self.offset

        self.lastoffset = self.offset
      , @interval
```

**Comment by lukefx** [comment]
This should be fixed in #72
